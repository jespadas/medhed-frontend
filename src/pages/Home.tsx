import Container from '@mui/material/Container';
import Grid from '@mui/material/Grid';
import React, { useEffect, useState } from 'react';
import { getHospitals } from '../api/hospital';
import { ManageStatus, SimpleCard } from '../components';
import { HospitalState } from '../models/hospitalState';

function Home() {
	const [state, setState] = useState<HospitalState>({
		status: 'loading',
		data: [],
	});
	const getData = async () => {
		try {
      setState({ status: 'loading', data: [] });
			const response = await getHospitals();
			setState({ status: 'success', data: response.data });
		} catch (error) {
			setState({ status: 'error', data: [] });
		}
	};

	useEffect(() => {
		getData();
	}, []);

	return (
		<ManageStatus status={state.status}>
			<Container maxWidth='lg'>
				<Grid container spacing={4}>
					{state.data.map((hospital) => (
						<Grid item lg={6} sm={4} xs={12} key={hospital.id}>
							<SimpleCard
								to={`detail/${hospital.id}`}
								title={hospital.name}
								description={hospital.address}
								image={hospital.image}
								handleButton={() => console.log('click')} id={0}							/>
						</Grid>
					))}
				</Grid>
			</Container>
		</ManageStatus>
	);
}

export default Home;
