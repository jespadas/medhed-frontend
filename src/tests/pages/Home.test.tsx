import React from 'react';
import { shallow } from "enzyme";
import Home from '../../pages/Home';

describe('Test in <Home />', () => {

    const wrapper = shallow(<Home />);

    test('Should be displayed correctly', () => {

        expect(wrapper).toMatchSnapshot();

    });

});