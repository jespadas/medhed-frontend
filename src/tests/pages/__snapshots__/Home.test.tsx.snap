// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <Home /> Should be displayed correctly 1`] = `
<ManageStatus
  status="loading"
>
  <ForwardRef(Container)
    maxWidth="lg"
  >
    <ForwardRef(Grid)
      container={true}
      spacing={4}
    />
  </ForwardRef(Container)>
</ManageStatus>
`;
